code/modules/mob/living/simple_animal/abnormality/aleph/distortedform.dm

```diff
@@ -0,0 +1,1728 @@
+
+/mob/living/simple_animal/hostile/abnormality/distortedform/FearEffectText(mob/affected_mob, level = 0)
+	level = num2text(clamp(level, 1, 5))
+	var/list/result_text_list = list(
+		"1" = list("There's no room for error here.", "My legs are trembling...", "Damn, it's scary."),
+		"2" = list("GODDAMN IT!!!!", "H-Help...", "I don't want to die!"),
+		"3" = list("What am I seeing...?", "I-I can't take it...", "I can't understand..."),
+		"4" = list("I'm so dead...", "That thing's just a monster!", "I need to get out of here!"),
+		"5" = list("Is that what it really looks like?", "It's over...", "I can’t even move my legs...")
+		)
+	return pick(result_text_list[level])
+
+//Work Mechanics
+/mob/living/simple_animal/hostile/abnormality/distortedform/AttemptWork(mob/living/carbon/human/user, work_type)
+	if(transformed) // distorted form jumpscare
+		UnTransform(TRUE)
+	return ..()
+
+/mob/living/simple_animal/hostile/abnormality/distortedform/WorktickFailure(mob/living/carbon/human/user)
+	var/list/damtypes = list(RED_DAMAGE, WHITE_DAMAGE, BLACK_DAMAGE, PALE_DAMAGE)
+	for(var/damagetype in damtypes) // take 4 of every damage type every failed tick
+		user.apply_damage(work_damage_amount, damagetype, null, user.run_armor_check(null, damagetype))
+
+/mob/living/simple_animal/hostile/abnormality/distortedform/SuccessEffect(mob/living/carbon/human/user, work_type, pe)
+	datum_reference.qliphoth_change(1)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/distortedform/FailureEffect(mob/living/carbon/human/user, work_type, pe)
+	addtimer(CALLBACK(src, .proc/CauseMelts), 10) //Delaying it prevents some bugs; call a meltdown on a bad result or failed stat check
+	return
+
+/mob/living/simple_animal/hostile/abnormality/distortedform/PostWorkEffect(mob/living/carbon/human/user, work_type, pe, work_time)
+	if(get_attribute_level(user, JUSTICE_ATTRIBUTE) < 100)
+		if(pe > datum_reference.neutral_boxes) // Not work failure, we don't need to call TWO melts
+			addtimer(CALLBACK(src, .proc/CauseMelts), 10)
+	return
+
+//Contained Mechanics
+/mob/living/simple_animal/hostile/abnormality/distortedform/proc/Transform()
+	if(!(IsContained(src)))
+		return
+	if(datum_reference.working) //If we are being worked on - transform later instead.
+		transform_timer = addtimer(CALLBACK(src, .proc/Transform), 3 MINUTES, TIMER_STOPPABLE & TIMER_OVERRIDE)
+		return
+	var/list/abno_list = AbnoListGen()
+	if(!LAZYLEN(abno_list))
+		return
+	var/datum/abnormality/abno_datum = pick(abno_list)
+	var/mob/living/simple_animal/hostile/abnormality/abno = abno_datum.current
+	soundloop1.stop()
+	soundloop2.stop()
+	new /obj/effect/temp_visual/distortedform_shift(get_turf(src))
+	name = abno.name
+	desc = abno.desc
+	icon = abno.icon
+	icon_state = abno.icon_state
+	pixel_x = abno.pixel_x
+	base_pixel_x = abno.base_pixel_x
+	pixel_y = abno.pixel_y
+	base_pixel_y = abno.base_pixel_y
```

Comment:
You can just set DF's transform to that of selected abno, it will do all of these(save for name and desc).
Also, add a check for dead abnos.

Reply:
I tried this, and it did not work as consistently. I may try another approach in the future.

Reply:
Added a check for dead abnos at least.